Gems/Atom/Asset/ImageProcessingAtom/Code/Source/Thumbnail/ImageThumbnail.cpp

```diff
@@ -81,15 +85,21 @@ namespace ImageProcessingAtom
             m_renderWait.release();
         }
 
-        void ImageThumbnail::OnCatalogAssetChanged([[maybe_unused]] const AZ::Data::AssetId& assetId)
+        void ImageThumbnail::OnCatalogAssetChanged(const AZ::Data::AssetId& assetId)
         {
             if (m_state == State::Ready && m_assetIds.find(assetId) != m_assetIds.end())
             {
-                m_state = State::Unloaded;
-                Load();
+                AZ::Data::AssetBus::MultiHandler::BusConnect(assetId);
             }
         }
```

Comment:
I don't really mind this but just wondering what the reasoning is behind listening to OnCatalogAssetChanged and then BusConnecting vs simply staying connected all the time?  Seems like it would simplify the code some

Reply:
I think the intention is to show loading icon as thumbnail once it detects any file changes. And start loading the new asset at same time. When loading is finished, it shows the new asset's thumbnail.

I think my change didn't show the loading icon until late. I should update the code.